File: beautifulDays_hacker_rank.rb
def beautifulDays(i, j, k)
    count = 0
    start = i
    while start <= j
        count += 1 if (start - "#{start}".reverse.to_i).abs % k == 0
        start += 1
    end
    count
end

p beautifulDays(20, 23, 6)

File: beautifulDays_hacker_rank.js
function beautifulDays(i, j, k) {
    console.log([i, j, k])
    let count = 0;
    for (let num = i; num <= j; num++) {
        let oppo = num.toString();
        oppo = parseInt(oppo.split('').reverse().join(''))
        let val = Math.abs(num - oppo);
        if (val % k == 0) {
            count += 1
        }
    }
    return count;
}